File: Backend/app/schemas/failure.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional
from .user import UserReference


class FailureCreate(BaseModel):
    description: str
    asset_id: Optional[int] = None
    component_id: Optional[int] = None
    severity: str = "MEDIUM"


class FailureRead(BaseModel):
    id: int
    description: str
    status: str
    severity: str
    asset_id: Optional[int] = None
    component_id: Optional[int] = None
    reported_by: int
    reported_date: datetime
    resolved_date: Optional[datetime] = None
    resolution_notes: Optional[str] = None
    created_at: datetime
    updated_at: Optional[datetime] = None

    class Config:
        from_attributes = True


class FailureReadWithUser(FailureRead):
    reported_by_user: Optional[UserReference] = None

    class Config:
        from_attributes = True


class FailureUpdate(BaseModel):
    description: Optional[str] = None
    # Use string types for consistency
    status: Optional[str] = None
    severity: Optional[str] = None
    resolved_date: Optional[datetime] = None
    resolution_notes: Optional[str] = None


class FailureWithWorkOrder(BaseModel):
    id: int
    description: str
    severity: str
    status: str
    reported_by: int
    reported_date: datetime
    resolved_date: Optional[datetime] = None
    asset_id: Optional[int] = None
    component_id: Optional[int] = None
    created_at: datetime
    updated_at: Optional[datetime] = None
    workorder_id: Optional[int] = None

    class Config:
        from_attributes = True

File: Frontend/zyro-web/src/app/types/failure.ts
/**
 * Tipos para fallos
 */

export interface FailureCreate {
  description: string;
  asset_id?: number;
  component_id?: number;
  severity?: string;
}

export interface FailureRead {
  id: number;
  description: string;
  status: string;
  severity: string;
  asset_id?: number;
  component_id?: number;
  reported_by: number;
  organization_id: number;
  reported_date: string;
  resolved_date?: string;
  resolution_notes?: string;
  created_at: string;
  updated_at?: string;
}

export interface FailureUpdate {
  description?: string;
  status?: string;
  severity?: string;
  resolved_date?: string;
  resolution_notes?: string;
}

export interface FailureWithWorkOrder extends FailureRead {
  workorder_id?: number;
}
